docs/source/theory.rst

```diff
@@ -0,0 +1,235 @@
+
+Then, compute the midpoint frequency :math:`\omega_{k, k+1} = \frac{\omega_k + \omega_{k+1}}{2}` of each existing frequency subinterval.
+The Simpson's rule approximation of the integrated interpolation error is given then as
+
+.. math::
+    \text{Total error} = \sum_k \frac{2(\omega_{k+1} - \omega_k)}{3} | \psi(\omega_{k, k+1}) - p_k(\omega_{k, k+1}) |,
+
+where the contribution from the interval endpoints has been removed since they by construction of the interpolant are identically zero.
+By iteratively adding the midpoint frequency from the interval with largest error, and recomputing the error with the two new intervals added,
+the error should gradually shrink, and one can terminate the iteration when reaching a desired tolerance.
+
+As an alternative to bisecting the subintervals using the arithmetic mean, as shown above, one could use the geometric mean instead:
+
+.. math::
+    \omega_{k, k+1} = \operatorname{sign}(\omega_k) \sqrt{\omega_k \omega_{k+1}} = e^{\frac{\log(\omega_k) + \log(\omega_{k+1})}{2}}.
+
+This can not be done for intervals containing zero, and requires Simpson's formula to be modified to
+
+.. math::
+    \text{Total error} = \sum_k \frac{2}{3} \log\left({\frac{\omega_{k+1}}{\omega_k}}\right) | \psi(\omega_{k, k+1}) - p_k(\omega_{k, k+1}) |\omega_{k, k+1}.
+
+The calculation steps are shown in :ref:`appendix-logscale-simpson`.
+One can also combine the two approaches, selecting either arithmetic or geometric bisection depending on the frequency.
+
+Regarding the determination of good frequency end points :math:`\omega_{min}, \omega_{max}`, one can incorporate this into the bisection algorithm by also allowing the intervals
+:math:`(-\infty, \omega_{min})` and :math:`(\omega_{max}, \infty)` to be bisected.
+This can be done by creating a phantom frequency by taking the sum or product of :math:`\omega_{min}` or :math:`\omega_{max}` and some constant,
+and using this phantom frequency for the bisection and error integration.
+
+
+Appendix
+--------
+
+.. _appendix-polynomial-examples:
+
+Appendix A - Integration of linear and PCHIP interpolants
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As mentioned in `Integrating the main integration range`_, when performing the Fourier integral using Filon's method, any interpolating polynomial can in principle be used.
+Here, we show the solution for a linear interpolant and a PCHIP interpolant. Derivation of these results are given in [N.Mounet-PhD]_.
+
+In both solutions,
+
+.. math::
+    \Delta_k = \omega_{k+1} - \omega_k
+
+denotes the length of the frequency interval. It is also worth noting that while e.g. :math:`p_k(\omega)` is written in the expressions,
+by construction, this is identical to :math:`\psi(\omega)` as well.
+There is in practice therefore not necessary to compute the interpolant (i.e. slopes and constants on each interval for linear interpolation),
+as knowing the function values is enough.
```

Comment:
I don't get what is meant here. First, `p_k` is not exactly `\psi` (it is its approximation), and second, you need the slopes in the PCHIP case.

Reply:
What I meant is that you don't actually need to compute e.g. the coefficients of the polynomial, that knowing the function values (and slopes for PCHIP) is sufficient. But I agree it is not clear, I will rephrase